app/guided.py

```diff
@@ -0,0 +1,388 @@
+
+import collections
+import copy
+import heapq
+import itertools
+import re
+
+import networkx as nx
+
+from app.objects.c_ability import Ability
+from app.objects.c_agent import Agent
+from app.objects.secondclass.c_goal import Goal
+
+
+FACT_REGEX = r'#{(.*?)}'
```

Comment:
This regex needs to be fixed. It parses out the string between #{ and }, but this leads to the input fact being parsed differently when there are filters on it. For example, "host.file.path" and "host.file.path[filters(technique=T1005,max=3)]" are interpreted as different facts in the graph, leading to disconnected components.